src/Libraries/Microsoft.Extensions.AI.Abstractions/TextToImage/TextToImageResponse.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Serialization;
+using Microsoft.Shared.Diagnostics;
+
+#pragma warning disable EA0011 // Consider removing unnecessary conditional access operators
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>Represents the result of an image generation request.</summary>
+[Experimental("MEAI001")]
+public class TextToImageResponse
+{
+    /// <summary>The content items in the generated text response.</summary>
+    private IList<AIContent>? _contents;
+
+    /// <summary>Initializes a new instance of the <see cref="TextToImageResponse"/> class.</summary>
+    [JsonConstructor]
+    public TextToImageResponse()
+    {
+    }
+
+    /// <summary>Initializes a new instance of the <see cref="TextToImageResponse"/> class.</summary>
+    /// <param name="contents">The contents for this response.</param>
+    /// <exception cref="ArgumentNullException"><paramref name="contents"/> is <see langword="null"/>.</exception>
+    public TextToImageResponse(IList<AIContent> contents)
+    {
+        _contents = Throw.IfNull(contents);
```

Comment:
This is throwing if the contents is null, but the equivalent setter is not. In the equivalent ChatResponse ctor, we have contents annotated to be nullable, should probably do the same thing here.